File: 02_ProgrammingFundamentals/_10_TextProcessing/_03_MoreExercises/_01_ExtractPersonInformation.java
package _02_ProgrammingFundamentals._10_TextProcessing._03_MoreExercises;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class _01_ExtractPersonInformation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int numberOfStrings = Integer.parseInt(scan.nextLine());

        // Add variables
        String regexName = "@(?<name>[A-Za-z]+)\\|";
        String regexAge = "#(?<age>[0-9]+)\\*";
        Pattern patterName = Pattern.compile(regexName);
        Pattern patternAge = Pattern.compile(regexAge);

        // Extract user information
        for (int i = 0; i < numberOfStrings; i++) {
            String newLine = scan.nextLine();

            Matcher matcherName = patterName.matcher(newLine);
            Matcher matcherAge = patternAge.matcher(newLine);

            String name = "";
            String age = "";

            if (matcherName.find()) {
                name = matcherName.group("name");
            }

            if (matcherAge.find()) {
                age = matcherAge.group("age");
            }

            if (name.length() > 0 && age.length() > 0) {
                System.out.printf("%s is %s years old.%n", name, age);
            }
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_08_TextProcessing/_03_MoreExercises/_01_ExtractPersonInformation/_01_ExtractPersonInformation.cs
using System;
using System.Text.RegularExpressions;

public class _01_ExtractPersonInformation
{
    public static void Main(string[] args)
    {
        // Read input value
        int numberOfStrings = int.Parse(Console.ReadLine());

        // Add variables
        string regexName = "@(?<name>[A-Za-z]+)\\|";
        string regexAge = "#(?<age>[0-9]+)\\*";
        Regex patterName = new Regex(regexName);
        Regex patternAge = new Regex(regexAge);

        // Extract user information
        for (int i = 0; i < numberOfStrings; i++)
        {
            string newLine = Console.ReadLine();

            Match matcherName = patterName.Match(newLine);
            Match matcherAge = patternAge.Match(newLine);

            string name = "";
            string age = "";

            if (matcherName.Success)
            {
                name = matcherName.Groups["name"].Value;
            }

            if (matcherAge.Success)
            {
                age = matcherAge.Groups["age"].Value;
            }

            if (name.Length > 0 && age.Length > 0)
            {
                Console.WriteLine($"{name} is {age} years old.");
            }
        }
    }
}
